Branch-SDK-Tests/BNCDeviceInfoTests.m

```diff
@@ -0,0 +1,145 @@
+//
+//  BNCDeviceInfoTests.m
+//  Branch-SDK-Tests
+//
+//  Created by Ernest Cho on 11/21/19.
+//  Copyright © 2019 Branch, Inc. All rights reserved.
+//
+
+#import <XCTest/XCTest.h>
+#import "BNCDeviceInfo.h"
+
+@interface BNCDeviceInfoTests : XCTestCase
+@property (nonatomic, strong, readwrite) BNCDeviceInfo *deviceInfo;
+@end
+
+@implementation BNCDeviceInfoTests
+
+- (void)setUp {
+    self.deviceInfo = [BNCDeviceInfo new];
+}
+
+- (void)tearDown {
+
+}
+
+- (void)testAppVersion {
+    // checks test app version
+    XCTAssert([@"1.1" isEqualToString:self.deviceInfo.applicationVersion]);
+}
+
+- (void)testBrandName {
+    XCTAssert([@"Apple" isEqualToString:self.deviceInfo.brandName]);
+}
+
+- (void)testModelName_Simulator {
+    XCTAssert([@"x86_64" isEqualToString:self.deviceInfo.modelName]);
+}
+
+//- (void)testModelName_iPhone7 {
+//    XCTAssert([@"iPhone9,3" isEqualToString:self.deviceInfo.modelName]);
+//}
+
+- (void)testOSName {
+    XCTAssertNotNil(self.deviceInfo.osName);
+    XCTAssert([self.deviceInfo.osName isEqualToString:[UIDevice currentDevice].systemName]);
+}
+
+- (void)testOSVersion {
+    XCTAssertNotNil(self.deviceInfo.osVersion);
+    XCTAssert([self.deviceInfo.osVersion isEqualToString:[UIDevice currentDevice].systemVersion]);
+}
+
+- (void)testOSBuildVersion {
+    XCTAssertNotNil(self.deviceInfo.osBuildVersion);
+}
+
+- (void)testExtensionType {
+    XCTAssert([@"FULL_APP" isEqualToString:self.deviceInfo.extensionType]);
+}
+
+- (void)testCpuType_Simulator {
+    NSNumber *expected = @(7);
+    XCTAssert([expected isEqualToNumber:self.deviceInfo.cpuType]);
```

Comment:
Most of these testing seems to be making assertion on default `[BNCDeviceInfo new];`

I think it would be easier to put them as a single test method.